server/src/main/java/org/opensearch/indices/replication/PrimaryShardReplicationSource.java

```diff
@@ -78,14 +80,33 @@ public void getSegmentFiles(
     ) {
         final Writeable.Reader<GetSegmentFilesResponse> reader = GetSegmentFilesResponse::new;
         final ActionListener<GetSegmentFilesResponse> responseListener = ActionListener.map(listener, r -> r);
+        // Few of the below assumptions and calculations are added for experimental release of segment replication feature in 2.3
+        // version.These will be changed in next release.
+
+        // Storing the size of files to fetch in bytes.
+        long sizeOfSegmentFiles = 0;
+        for (int i = 0; i < filesToFetch.size(); i++) {
+            sizeOfSegmentFiles += filesToFetch.get(i).length();
+        }
+        // Making sure files size is in correct format to perform time calculation.
+        sizeOfSegmentFiles = Math.abs(sizeOfSegmentFiles);
+        // Maximum size of files to fetch (segment files), that can be processed in 1 minute for a m5.xlarge machine.
+        long baseSegmentFilesSize = 300000000;
```

Comment:
1. Can you share steps performed to identify this baseline ? 
2. This baseline is calculated for `m5.xlarge` instance type, which means all instance types with lower throughput may fail persistently ? Can we relax(reduce) this baseline to smaller value to accomodate lower grade instance types. It doesn't hurt to reduce this limit for high grade instance type, only side-effect is that we estimate higher number of minutes but it will complete faster. But, it will fix timeout issues on lower grade machines.

Reply:
+1.   This feels a bit high, maybe bump by 1M for every 100mb? This value should be toggleable with a setting, we can shoot for that in next release.

Reply:
Steps to identify baseline:
-> I started a 3 node cluster and ran opensearch-benchmarks with SO dataset. I logged files to fetch size and check the logs later for every replication event. After checking all replication event logs i observed the maximum files to fetch size that has finished within 1min time was 330mb, so I took 330mb as baseline. Later after this change no timeouts happened.

->As I mentioned above that after observing logs I put baseline as 330mb, but as both @dreamer-89 and @mch2 mentioned it makes sense to reduce the baseline, so that lower throughput machines don't have a timeout. I will use bump by 1min for every 100 mb as @mch2 suggested.